## Database

### [HGNC](https://www.genenames.org): is responsible for approving unique symbols and names for human loci, including protein coding genes, ncRNA genes and pseudogenes, to allow unambiguous scientific communication.


## Websites

#### [Segmental duplications (hg38)](ftp://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/genomicSuperDups.txt.gz)
#### [self-chains (hg38)](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/chainSelf.txt.gz)
#### [Pfam domains (hg38)](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/ucscGenePfam.txt.gz)
#### [hg38 gtf](https://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz)
#### [gnomAD](https://storage.googleapis.com/gnomad-public/release/2.0.1/vcf/exomes/gnomad.exomes.r2.0.1.sites.vcf.gz)
#### [CADD files for both indels and SNPs (hg38)](http://krishna.gs.washington.edu/download/CADD/v1.5/GRCh38/)
#### [MPC](ftp://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/fordist_constraint_official_mpc_values.txt.gz)
#### [whole-exome MTR file](http://mtr-viewer.mdhs.unimelb.edu.au:8079/mtrflatfile_1.0.txt.gz)
#### [REVEL file](https://rothsj06.u.hpc.mssm.edu/revel/revel_all_chromosomes.csv.zip)
#### [pLI file](ftp://ftp.broadinstitute.org/pub/ExAC_release/release1/manuscript_data/forweb_cleaned_exac_r03_march16_z_data_pLI.txt.gz)
#### [ClinVar VCF file](ftp://ftp.ncbi.nih.gov/pub/clinvar/vcf_GRCh37/archive_2.0/2017/clinvar_20170802.vcf.gz)
#### [de novo variants on s3 server](https://s3.us-east-2.amazonaws.com/pathoscore-data/samocha/samochadenovo.xlsx)