# Bioinformatic-resources

## Trio & family

* [trio binning](https://obj.umiacs.umd.edu/marbl_publications/triobinning/index.html)
* [denovogear](https://github.com/denovogear/denovogear): A program to detect denovo-variants using next-generation sequencing data. 
* [VBT-TrioAnalysis](https://github.com/sbg/VBT-TrioAnalysis)
* [fitDNM](https://github.com/jeremymcrae/fitDNM): Enrichment of de novo mutations within genes
* [denovo-db](http://denovo-db.gs.washington.edu/)
* [denovodb v.1.5 documentation](http://denovo-db.gs.washington.edu/denovo-db.v.1.5.pdf)
* [DenovolyzeR](http://denovolyzer.org)
* [TrioDeNovo](http://genome.sph.umich.edu/wiki/Triodenovo)


## Quality control
* [adapterremoval](https://github.com/MikkelSchubert/adapterremoval): rapid adapter trimming, identification, and read merging

## Tools with bam
* [Alfred](https://tobiasrausch.com/alfred/): BAM alignment statistics, feature counting and feature annotation
* [bamkit](https://github.com/hall-lab/bamkit): Tools for common BAM file manipulations
* [bam-readcount](https://github.com/genome/bam-readcount): count DNA sequence reads in BAM files
* [bamtools](https://github.com/pezmaster31/bamtools)
* [biobambam2](https://github.com/gt1/biobambam2): Tools for early stage alignment file processing
* [mosdepth](https://github.com/arq5x/mosdepth): fast BAM/CRAM depth calculation for WGS, exome, or targetted sequencing.
* [VariantBam](https://github.com/walaj/VariantBam): Filtering and profiling of next-generational sequencing data using region-specific rules


## Alignment (Illumina)
* [bwa](https://github.com/lh3/bwa): Burrow-Wheeler Aligner for short-read alignment (see minimap2 for long-read alignment)


## fusion gene
* [arriba](https://github.com/suhrig/arriba): Fast and accurate gene fusion detection from RNA-Seq data
* [FuSeq](https://github.com/nghiavtr/FuSeq): A fast detection of fusion genes from paired-end RNA-seq data
* [GeneFuse](https://github.com/OpenGene/GeneFuse): Gene fusion detection and visualization
* [fusioncatcher](https://github.com/ndaniel/fusioncatcher): Finder of Somatic Fusion Genes in RNA-seq data
* [STAR-Fusion](https://github.com/STAR-Fusion/STAR-Fusion): STAR-Fusion codebase
* [STAR-Fusion-Tutorial](https://github.com/STAR-Fusion/STAR-Fusion-Tutorial): Tutorial for STAR-Fusion, FusionInspector, and de novo reconstruction of fusion transcripts using Trinity


## 16S rRNA resources

### data format
* [Metadata mapping files](http://qiime.org/documentation/file_formats.html#metadata-mapping-files)
* [QIIME File Formats](http://qiime.org/documentation/index.html)
* [16S ribosomal RNA](https://en.wikipedia.org/wiki/16S_ribosomal_RNA)


### 16S rRNA gene database
* [RDP](https://rdp.cme.msu.edu/): RDP provides quality-controlled, aligned and annotated Bacterial and Archaeal 16S rRNA sequences
* [SILVA](https://www.arb-silva.de/): SILVA provides comprehensive, quality checked and regularly updated datasets of aligned small (16S/18S, SSU) and large subunit (23S/28S, LSU) ribosomal RNA (rRNA) sequences
* [GreenGene](http://www.greengene.com/): Greengenes is a quality controlled, comprehensive 16S reference database and taxonomy based on a de novo phylogeny that provides standard operational taxonomic unit sets
* [rrnDB](https://rrndb.umms.med.umich.edu/): A searchable database documenting variation in ribosomal RNA operons (rrn) in Bacteria and Archaea
* [EzTaxon-e](https://www.ezbiocloud.net/): It contains comprehensive 16S rRNA gene sequences of taxa with valid names as well as sequences of uncultured taxa


### Tools
* [dada2](https://github.com/benjjneb/dada2): Accurate sample inference from amplicon data with single nucleotide resolution


## Metagenome

### Tools

* [microbiomeGWAS](https://github.com/lsncibb/microbiomeGWAS)
* [metaSNV](http://metasnv.embl.de/tutorial.html)
* [Metagenomics](http://www.metagenomics.wiki/tools/annotation)
* [metag-rev-sup](https://github.com/chrisquince/metag-rev-sup)
* [PanPhlAn](https://bitbucket.org/CibioCM/panphlan/wiki/Home)
* [MetaBAT](https://bitbucket.org/berkeleylab/metabat/wiki/Home)
* [Galaxy ToolShed](https://galaxyproject.org/toolshed/)
* [multi-metagenome](https://github.com/MadsAlbertsen/multi-metagenome): Genome sequences of rare, uncultured bacteria
* [srst2](https://github.com/katholt/srst2): Short Read Sequence Typing for Bacterial Pathogens
* [ariba](https://github.com/sanger-pathogens/ariba/wiki): Antimicrobial Resistance Identification By Assembly
* [abricate](https://github.com/tseemann/abricate): Mass screening of contigs for antimicrobial and virulence genes
* [snippy](https://github.com/tseemann/snippy): Rapid bacterial SNP calling and core genome alignments
* [mag](https://github.com/nf-core/mag): Assembly and binning of metagenomes
* [decontam](https://github.com/benjjneb/decontam): Simple statistical identification and removal of contaminants in marker-gene and metagenomics sequencing data
* [MetaCarvel](https://github.com/marbl/MetaCarvel): A scaffolder for metagenomes
* [MetaXcan](https://github.com/hakyimlab/MetaXcan)


### pipeline

* [EBAME18](https://github.com/LomanLab/EBAME18): Long read metagenomics practical at EBAME18
* [bac-ngs-book](https://github.com/indexofire/bac-ngs-book): 病原微生物高通量测序数据分析笔记
* [Zhang2019NBT](https://github.com/YongxinLiu/Zhang2019NBT)
* [Microbiota](https://github.com/microbiota)




## Single cell transcriptome
* [BISCUIT_SingleCell_IMM_ICML_2016](https://github.com/sandhya212/BISCUIT_SingleCell_IMM_ICML_2016): R Codebase for BISCUIT: Infinite Mixture Model to cluster and impute single cells.
* [cisTopic](https://github.com/aertslab/cisTopic): Probabilistic modelling of cis-regulatory topics from single cell epigenomics data
* [CONICS](https://github.com/diazlab/CONICS):  COpy-Number analysis In single-Cell RNA-Sequencing
* [DoubletDetection](https://github.com/JonathanShor/DoubletDetection): Doublet detection in single-cell RNA-seq data.
* [dropSeqPipe](https://github.com/Hoohm/dropSeqPipe): A SingleCell RNASeq pre-processing pipeline built on snakemake
* [HoneyBADGER](https://github.com/JEFworks/HoneyBADGER): HMM-integrated Bayesian approach for detecting CNV and LOH events from single-cell RNA-seq data
* [ImmuneResistance](https://github.com/livnatje/ImmuneResistance): Single-cell RNA-seq of melanoma ecosystems reveals sources of T cell exclusion linked to immunotherapy clinical outcomes
* [inferCNV](https://github.com/broadinstitute/inferCNV): Inferring CNV from Single-Cell RNA-Seq
* [SAVER](https://github.com/mohuangx/SAVER): Single-cell RNA-seq Gene Expression Recovery
* [scanpy](https://github.com/theislab/scanpy): Single-Cell Analysis in Python. Scales to >1M cells. http://scanpy.rtfd.io
* [scde](https://github.com/hms-dbmi/scde): R package for analyzing single-cell RNA-seq data 
* [scImpute](https://github.com/Vivianstats/scImpute): Accurate and robust imputation of scRNA-seq data
* [scell](https://github.com/diazlab/scell): Single-CELL rna-seq analysis software 
* [scg_lib_structs](https://github.com/Teichlab/scg_lib_structs): Collections of library structure and sequence of popular single cell genomic methods
* [single_cell_portal_core](https://github.com/broadinstitute/single_cell_portal_core): Rails/Docker application for the Broad Institute single cell RNA-seq data portal
* [single-cell-pseudotime](https://github.com/agitter/single-cell-pseudotime): An overview of algorithms for estimating pseudotime in single-cell RNA-seq data
* [single-cell-tutorial](https://github.com/theislab/single-cell-tutorial)
* [SingleR](https://github.com/dviraran/SingleR): Single-cell RNA-seq cell types Recognition
* [scRNA-tools](https://github.com/Oshlack/scRNA-tools): Table of software for the analysis of single-cell RNA-seq data.
* [seurat](https://github.com/satijalab/seurat): R toolkit for single cell genomics
* [snATAC](https://github.com/r3fang/snATAC): Ren Lab in-house dual-barcode single nucleus ATAC-seq (snATAC-seq) analysis pipeline
* [STREAM_atac](https://github.com/pinellolab/STREAM_atac): Single-cell Trajectories Reconstruction, Exploration And Mapping of single-cell data. Preprocessing steps for single cell atac-seq data
* [STREAM](https://github.com/pinellolab/STREAM): Single-cell Trajectories Reconstruction, Exploration And Mapping of single-cell data
* [tenx](https://github.com/sansomlab/tenx): Pipelines for the analysis of 10x single-cell RNA-sequencing data
* [awesome-single-cell](https://github.com/seandavi/awesome-single-cell)
* [scPipe](https://github.com/LuyiTian/scPipe): a pipeline for single cell RNA-seq data analysis
* [Linnarsson Lab Single-cell analysis of mouse cortex](http://linnarssonlab.org/cortex)
* [Human MTG single nucleus RNA-seq data](http://celltypes.brain-map.org/download)
* [scMerge](https://github.com/SydneyBioX/scMerge):Statistical technique for removing unwanted variation from multiple scRNA-seq datasets
* [scRNA-seq-workshop-Fall-2018](https://github.com/harvardinformatics/scRNA-seq-workshop-Fall-2018)
* [SoupX](https://github.com/constantAmateur/SoupX): R package to quantify and remove cell free mRNAs from droplet based scRNA-seq data




## Nanopore

### pipeline
* [NanoDJ](https://github.com/genomicsITER/NanoDJ): A Dockerized Jupyter Notebook for Interactive Oxford Nanopore MinION Sequence Manipulation and Genome Assembly
* [gnomad-sv-pipeline](https://github.com/talkowski-lab/gnomad-sv-pipeline): Code and custom scripts relevant to gnomAD-SV (Collins*, Brand*, et al., 2019)
* [sv-benchmark](https://github.com/PacificBiosciences/sv-benchmark): Public Benchmark of Long-Read Structural Variant Caller on PacBio CCS HG002 Data
* [Pomoxis](https://github.com/nanoporetech/pomoxis): comprises a set of basic bioinformatic tools tailored to nanopore sequencing
* [Nanoflow](https://github.com/zhaoc1/nanoflow): a NANOpore sequencing data bioinformatics workFLOW
* [Scrappie](https://github.com/nanoporetech/scrappie): a technology demonstrator for the Oxford Nanopore Research Algorithms group
* [wub](https://github.com/nanoporetech/wub): Tools and software library developed by the ONT Applications group
* [nanopore-scripts](https://github.com/arq5x/nanopore-scripts)
* [nano-snakemake](https://github.com/wdecoster/nano-snakemake): A snakemake pipeline for SV analysis from nanopore genome sequencing
* [pipeline-pinfish-analysis](https://github.com/nanoporetech/pipeline-pinfish-analysis): Pipeline for annotating genomes using long read transcriptomics data with pinfish
* [hpv_minION_analysis](https://github.com/edawson/hpv_minION_analysis): Contains scripts used to analyze HPV samples sequenced on ONT minIONs.


### quality control
* [fast5seek](https://github.com/mbhall88/fast5seek): Subset of fast5 files contained in a fastq, BAM, or SAM file
* [albacore](https://github.com/dvera/albacore): Dockerfile for the Albacore basecaller from Oxford Nanopore
* [Basecalling-comparison](https://github.com/rrwick/Basecalling-comparison): A comparison of different Oxford Nanopore basecallers
* [npBarcode](https://github.com/hsnguyen/npBarcode): Demultiplex barcoded Oxford Nanopore sequencing
* [npReader](https://github.com/mdcao/npReader): Real-time extraction and analysis Oxford Nanopore sequencing data
* [nanopore adapters](https://github.com/rrwick/Porechop#known-adapters)
* [NanoFilt](https://github.com/wdecoster/nanofilt): https://github.com/wdecoster/nanofilt
* [Deepbinner](https://github.com/rrwick/Deepbinner): a signal-level demultiplexer for Oxford Nanopore reads
* [Porechop](https://github.com/rrwick/Porechop): adapter trimmer for Oxford Nanopore reads
* [poretools](https://github.com/arq5x/poretools): a toolkit for working with Oxford nanopore data
* [NanoPlot](https://github.com/wdecoster/NanoPlot): Plotting scripts for long read sequencing data
* [Nanopolish](https://github.com/jts/nanopolish)
* [nanoQC](https://github.com/wdecoster/nanoQC): Quality control tools for nanopore sequencing data
* [NanoR](https://github.com/davidebolo1993/NanoR): R package for user-friendly analysis and comparison of ONT data
* [pomoxis](https://github.com/nanoporetech/pomoxis): Analysis components from Oxford Nanopore Research
* [poretools document](http://poretools.readthedocs.io/en/latest/)
* [poretools github](https://github.com/arq5x/poretools): a toolkit for working with Oxford nanopore data
* [qcat](https://github.com/nanoporetech/qcat): qcat is Python command-line tool for demultiplexing Oxford Nanopore reads from FASTQ files
* [pycoQC](https://github.com/a-slide/pycoQC): pycoQC computes metrics and generates Interactive QC plots from the sequencing summary report generated by Oxford Nanopore technologies basecaller (Albacore/Guppy)




### Assembly
* [NovoGraph](https://github.com/NCBI-Hackathons/NovoGraph): building whole genome graphs from long-read-based de novo assemblies
* [wtdbg2](https://github.com/ruanjue/wtdbg2): A fuzzy Bruijn graph approach to long noisy reads assembly
* [smartdenovo](https://github.com/ruanjue/smartdenovo): Ultra-fast de novo assembler using long noisy reads
* [MECAT2](https://github.com/xiaochuanle/MECAT2)
* [quickmerge](https://github.com/mahulchak/quickmerge): A simple and fast metassembler and assembly gap filler designed for long molecule based assemblies.
* [npGraph](https://github.com/hsnguyen/assembly):  Resolve assembly graph in real-time using nanopore data
* [Canu](https://github.com/marbl/canu)



###  Variants
* [Longshot](https://github.com/pjedge/longshot): diploid SNV caller for error-prone reads
* [NanoSatellite](https://github.com/arnederoeck/NanoSatellite): Dynamic time warping of Oxford Nanopore squiggle data to characterize tandem repeats



### Methylation
* [deepsignal](https://github.com/bioinfomaticsCSU/deepsignal): Detecting methylation using signal-level features from Nanopore sequencing reads
* [tombo](https://github.com/nanoporetech/tombo): a suite of tools primarily for the identification of modified nucleotides from raw nanopore sequencing data
* [DeepMod](https://github.com/WGLab/DeepMod): a deep-learning tool for genomic-scale, strand-sensitive and single-nucleotide based detection of DNA modifications
* [nanopore-methylation](https://github.com/YaoLi3/nanopore-methylation)
* [mCaller](https://github.com/al-mcintyre/mCaller): A python program to call methylation (m6A in DNA) from nanopore signal data



### Mapping tools

* [graphmap](https://github.com/isovic/graphmap): A highly sensitive and accurate mapper for long, error-prone reads


### simulator
* [NanoSim](https://github.com/bcgsc/NanoSim): Nanopore sequence read simulator
* [DeepSimulator](https://github.com/lykaust15/DeepSimulator): The first deep learning based Nanopore simulator which can simulate the process of Nanopore sequencing.



### data
* [NA12878 nanopore](https://github.com/nanopore-wgs-consortium/NA12878/blob/master/nanopore-human-genome/rel_3_4.md)
* [PromethION WGS data of NA19240](https://www.ebi.ac.uk/ena/data/view/PRJEB26791)
* [nanopore NA12878](https://github.com/nanopore-wgs-consortium/NA12878)


### Transcriptome
* [dRNA-paper-scripts](https://github.com/nanoporetech/dRNA-paper-scripts): Highly parallel direct RNA sequencing on an array of nanopores(https://www.nature.com/articles/nmeth.4577)
* [pinfish](https://github.com/nanoporetech/pinfish):Tools to annotate genomes using long read transcriptomics data
* [flair](https://github.com/BrooksLabUCSC/flair): Full-Length Alternative Isoform analysis of RNA



## Pacbio data
* [PacBioEDA](https://github.com/TomSkelly/PacBioEDA): Python scripts for Exploratory Data Analysis of Pacific Biosciences sequence data
* [GenomicConsensus](https://github.com/PacificBiosciences/GenomicConsensus): PacBio® variant and consensus caller
* [pbalign](https://github.com/PacificBiosciences/pbalign): pbalign maps PacBio reads to reference sequences and saves alignments to a BAM file
* [pbmm2](https://github.com/PacificBiosciences/pbmm2): A minimap2 frontend for PacBio native data formats 


## Assembly genome

### Assembly with short reads
* [w2rap-contigger](https://github.com/bioinfologics/w2rap-contigger): An Illumina PE genome contig assembler, can handle large (17Gbp) complex (hexaploid) genomes.
* [w2rap](https://github.com/bioinfologics/w2rap): WGS (Wheat) Robust Assembly Pipeline
* [GFA-spec](https://github.com/GFA-spec/GFA-spec): Graphical Fragment Assembly (GFA) Format Specification
* [HapCUT2](https://github.com/vibansal/HapCUT2): software tools for haplotype assembly from sequence data
* [masurca](https://github.com/alekseyzimin/masurca): MaSuRCA Genome Assembler Quick Start Guide
* [minia](https://github.com/GATB/minia): Minia is a short-read assembler based on a de Bruijn graph
* [npScarf](https://github.com/mdcao/npScarf): Scaffold and Complete assemblies in real-time fashion
* [redundans](https://github.com/lpryszcz/redundans): Redundans is a pipeline that assists an assembly of heterozygous/polymorphic genomes.
* [Scaff10X](https://github.com/wtsi-hpag/Scaff10X): Pipeline for scaffolding and breaking a genome assembly using 10x genomics linked-reads
* [SDA](https://github.com/mvollger/SDA): Segmental Duplication Assembler (SDA)
* [shovill](https://github.com/tseemann/shovill): Faster SPAdes assembly of Illumina reads
* [SOAPdenovo2]( https://sourceforge.net/projects/soapdenovo2/files/SOAPdenovo2/)


### Assembly with long reads
* [FALCON-Phase](https://github.com/phasegenomics/FALCON-Phase): FALCON-Phase integrates PacBio long-read assemblies with Phase Genomics Hi-C data to create phased, diploid, chromosome-scale scaffolds
* [wtdbg2](https://github.com/ruanjue/wtdbg2): A fuzzy Bruijn graph approach to long noisy reads assembly
* [DBG2OLC](https://github.com/yechengxi/DBG2OLC): The genome assembler that reduces the computational time of human genome assembly from 400,000 CPU hours to 2,000 CPU hours, utilizing long erroneous 3GS sequencing reads and short accurate NGS sequencing reads.
* [Flye](https://github.com/fenderglass/Flye): Fast and accurate de novo assembler for single molecule sequencing reads
* PBcR (http://wgs-assembler.sourceforge.net/wiki/index.php/PBcR)
* [SALSA](https://github.com/machinegun/SALSA): A tool to scaffold long read assemblies with Hi-C data
* [smartdenovo](https://github.com/ruanjue/smartdenovo): Ultra-fast de novo assembler using long noisy reads
* [NovoGraph](https://github.com/NCBI-Hackathons/NovoGraph): Genome Graph of Long-read De Novo Assemblies


### fill gap && polish
* [quickmerge](https://github.com/mahulchak/quickmerge): A simple and fast metassembler and assembly gap filler designed for long molecule based assemblies.
* [PBJelly](https://github.com/alvaralmstedt/Tutorials/wiki/Gap-closing-with-PBJelly): Gap-closing-with-PBJelly
* [GapCloser](http://soap.genomics.org.cn/about.html)

## Assembly transcriptome

* [Corset](https://github.com/Oshlack/Corset): Software for clustering de novo assembled transcripts and counting overlapping reads


## Variants

### Somatic variants

* [ascatNgs](https://github.com/cancerit/ascatNgs): Somatic copy number analysis using paired end wholegenome sequencing
* [needlestack](https://github.com/IARCbioinfo/needlestack): Multi-sample somatic variant caller
* [seurat](https://github.com/tgen/seurat): Tumor-Normal Variant Caller 
* [facets](https://github.com/mskcc/facets): Algorithm to implement Fraction and Copy number Estimate from Tumor/normal Sequencing.
* [Shimmer](https://github.com/nhansen/Shimmer): a software package for the characterization of genetic differences between two very similar samples, e.g., a tumor sample and its matched normal tissue sample
* [neusomatic](https://github.com/bioinform/neusomatic): Deep convolutional neural networks for accurate somatic mutation detection
* [Pisces](https://github.com/Illumina/Pisces): Somatic and germline variant caller for amplicon data.
* [deTiN](https://github.com/broadinstitute/deTiN): DeTiN is designed to measure tumor-in-normal contamination and improve somatic variant detection sensitivity when using a contaminated matched control.
* [DeepSVR](https://github.com/griffithlab/DeepSVR): a machine learning model approach to somatic variant refinement
* [somaticseq](https://github.com/bioinform/somaticseq): An ensemble approach to accurately detect somatic mutations using SomaticSeq
* [MuSiC2](https://github.com/ding-lab/MuSiC2): identifying mutational significance in cancer genomes


### Germline variants

* [benchmarking germline small-variant calls](https://github.com/ga4gh/benchmarking-tools): Repository for the GA4GH Benchmarking Team work developing standardized benchmarking methods for germline small variant calls
* [vt](https://github.com/atks/vt): A tool set for short variant discovery in genetic sequence data
* [dna-seq-gatk-variant-calling](https://github.com/snakemake-workflows/dna-seq-gatk-variant-calling): This Snakemake pipeline implements the GATK best-practices workflow
* [deepvariant](https://github.com/google/deepvariant): an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA sequencing data
* [speedseq](https://github.com/hall-lab/speedseq): A flexible framework for rapid genome analysis and interpretation
* [vg](https://github.com/vgteam/vg#vg): tools for working with genome variation graphs
* [GEMINI](https://github.com/arq5x/gemini): integrative exploration of genetic variation and genome annotations



### Haplotype &  phase
* [Complete assembly of parental haplotypes with trio binning](https://obj.umiacs.umd.edu/marbl_publications/triobinning/index.html)
* [Haplotype Reference Consortium (HRC)](http://www.haplotype-reference-consortium.org/)
* [HipSTR](https://github.com/tfwillems/HipSTR): Genotype and phase short tandem repeats using Illumina whole-genome sequencing data
* [Eagle](http://www.hsph.harvard.edu/alkes-price/software/): https://www.nature.com/articles/ng.3679



### Imputation
* [SHAPEIT](http://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html)
* [UK Biobank phasing and imputation documentation (including brief description of SHAPEIT3)](http://biobank.ctsu.ox.ac.uk/crystal/docs/impute_ukb_v1.pdf)
* [Beagle](http://faculty.washington.edu/browning/beagle/beagle.html)


### LD
* [PopLDdecay](https://github.com/BGI-shenzhen/PopLDdecay): a fast and effective tool for linkage disequilibrium decay analysis based on variant call format(VCF) files
* [emeraLD](https://github.com/statgen/emeraLD): tools to efficiently retrieve and calculate LD
* [ngsLD](https://github.com/fgvieira/ngsLD): Calculation of pairwise Linkage Disequilibrium (LD) under a probabilistic framework
* [LD Hub](http://ldsc.broadinstitute.org/ldhub/) 
* [LDSC and associated files](https://github.com/bulik/ldsc/)
* [abstar](https://github.com/briney/abstar): VDJ assignment and antibody sequence annotation. Scalable from a single sequence to billions of sequences.





## Structural variants

### SV Caller for third generation sequences
* [svim](https://github.com/eldariont/svim): Structural Variant Identification Method using Long Reads
* [SURVIVOR](https://github.com/fritzsedlazeck/SURVIVOR): Toolset for SV simulation, comparison and filtering
* [Sniffles](https://github.com/fritzsedlazeck/Sniffles): Structural variation caller using third generation sequencing
* [NanoSV](https://github.com/mroosmalen/nanosv): SV caller for nanopore data
* [smrtsv2](https://github.com/EichlerLab/smrtsv2): long read structural variant caller
* [pbsv](https://github.com/PacificBiosciences/pbsv): PacBio structural variant (SV) calling and analysis tools
* [Picky](https://github.com/TheJacksonLaboratory/Picky): Structural Variants Pipeline for Long Reads


### SV with illumina data
* [svtyper](https://github.com/hall-lab/svtyper): Bayesian genotyper for structural variants
* [lumpy-sv](https://github.com/arq5x/lumpy-sv): a general probabilistic framework for structural variant discovery
* [parliament2](https://github.com/dnanexus/parliament2): Runs a combination of tools to generate structural variant calls on whole-genome sequencing data
* [delly](https://github.com/dellytools/delly): Structural variant discovery by integrated paired-end and split-read analysis
* [manta](https://github.com/Illumina/manta): Structural variant and indel caller for mapped sequencing data
* [parliament2](https://github.com/dnanexus/parliament2): Runs a combination of tools to generate structural variant calls on whole-genome sequencing data
* [SV2](https://github.com/dantaki/SV2): Support Vector Structural Variation Genotyper
* [pindel](https://github.com/genome/pindel):  identify the breakpoints of these variants from paired-end short reads
* [MetaSV](https://bioinform.github.io/metasv/): An accurate and integrative structural-variant caller for next generation sequencing
* [svaba](https://github.com/walaj/svaba): Structural variation and indel detection by local assembly
* [wham](https://github.com/zeeev/wham): Structural variant detection and association testing
* [gridss](https://github.com/PapenfussLab/gridss): Genomic Rearrangement IDentification Software Suite
* [breakdancer](https://github.com/genome/breakdancer): SV detection from paired end reads mapping
* [SVenX](https://github.com/vborjesson/SVenX): Pipeline for SV detection using 10X genomics data
* [paragraph](https://github.com/Illumina/paragraph): Graph realignment tools for structural variants
* [svtools](https://github.com/hall-lab/svtools): Tools for processing and analyzing structural variants

### CNV

* [Control-FREE](http://boevalab.com/FREEC/): a tool for assessing copy number and allelic content using next generation sequencing data
* [canvas](https://github.com/Illumina/canvas): Canvas Copy Number Variant Caller
* [CNVnator](https://github.com/abyzovlab/CNVnator): a tool for CNV discovery and genotyping from depth-of-coverage by mapped reads
* [cnv_facets](https://github.com/dariober/cnv_facets): Somatic copy variant caller (CNV) for next generation sequencing
* [CNV-Visualizer](https://github.com/NabaviLab/CNV-Visualizer): Visualizing Copy Number Variations
* [facets](https://github.com/mskcc/facets): Algorithm to implement Fraction and Copy number Estimate from Tumor/normal Sequencing.
* [cnvkit](https://github.com/etal/cnvkit): Copy number variant detection from targeted DNA sequencing
* [ADTEx](http://adtex.sourceforge.net/tutorial.html): detect somatic copy number variations (CNVs) 
* [NGSEPcore](https://github.com/NGSEP/NGSEPcore): an integrated framework for analysis of high throughput sequencing (HTS) reads. The main functionality of NGSEP is the variants detector, which allows to make integrated discovery and genotyping of Single Nucleotide Variants (SNVs), insertions, deletions, and genomic regions with copy number variation (CNVs)
* [aCNViewer](https://github.com/FJD-CEPH/aCNViewer): Comprehensive genome-wide visualization of absolute copy number and copy neutral variations
* [cancerTitanCNA](https://github.com/gavinha/TitanCNA): Analysis of subclonal copy number alterations (CNA) and loss of heterozygosity (LOH)


### CNV workflow
* [TCAG-WGS-CNV-workflow](https://github.com/bjtrost/TCAG-WGS-CNV-workflow): Scripts involved in our workflow for detecting CNVs from WGS data using read depth-based methods
* [gatk4-somatic-cnvs](https://github.com/gatk-workflows/gatk4-somatic-cnvs)
* [Call somatic copy number variants using GATK4 CNV](https://gatkforums.broadinstitute.org/gatk/discussion/9143/how-to-call-somatic-copy-number-variants-using-gatk4-cnv)



### General
* [svaba](https://github.com/walaj/svaba): Structural variation and indel detection by local assembly
* [truvari](https://github.com/spiralgenetics/truvari): Structural variant comparison tool for VCFs
* [smoove](https://github.com/brentp/smoove): structural variant calling and genotyping with existing tools, but, smoothly
* [sv-pipeline](https://github.com/hall-lab/sv-pipeline): Pipeline for structural variation detection in cohorts
* [svtools](https://github.com/hall-lab/svtools): Tools for processing and analyzing structural variants
* [samplot](https://github.com/ryanlayer/samplot): Plot structural variant signals from many BAMs and CRAMs
* [svviz2](https://github.com/nspies/svviz2): visual evaluation of read support for structural variation


### SV annotation

* [AnnotSV](https://lbgi.fr/AnnotSV/): Annotation and Ranking of Human Structural Variations
* [Nirvana](https://github.com/Illumina/Nirvana): Nirvana provides clinical-grade annotation of genomic variants (SNVs, MNVs, insertions, deletions, indels, and SVs (including CNVs)
* [StructuralVariantAnnotation](https://github.com/PapenfussLab/StructuralVariantAnnotation): R package designed to simplify structural variant analysis


### SV (CNV) annotation database

* [RefSeq GRCh38](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/refGene.txt.gz)
* [Deciphering Developmental Disorders (DDD) Study](https://www.ddduk.org/)
* [OMIM (Online Mendelian Inheritance in Man)](https://www.omim.org/): https://omim.org/downloads/
* [gnomAD](https://gnomad.broadinstitute.org/)
* [GeneHancer](https://www.genecards.org/)
* [Exome Aggregation Consortium (ExAC)](http://exac.broadinstitute.org/)
* [ENCODE: Encyclopedia of DNA Elements](https://www.encodeproject.org/)
* [dbVar](https://www.ncbi.nlm.nih.gov/dbvar)
* [dbvar download](https://ftp.ncbi.nlm.nih.gov/pub/dbVar/data/Homo_sapiens/by_assembly/GRCh38/vcf/)
* [American College of Medical Genetics and Genomics (ACMG)](https://www.acmg.net/)
* [VarCards](http://varcards.biols.ac.cn/): an integrated genetic and clinical database for coding variants in the human genome




## GWAS
* [SNPTEST](https://mathgen.stats.ox.ac.uk/genetics_software/snptest/snptest.html):  a program for the analysis of single SNP association in genome-wide studies.
* [GTOOL](http://www.well.ox.ac.uk/~cfreeman/software/gwas/gtool.html): a program for transforming sets of genotype data for use with the programs SNPTEST and IMPUTE.
* [fcGENE](https://sourceforge.net/projects/fcgene/): A Versatile Tool for Processing and Transforming SNP Datasets.
* [Fast GWAS download script](https://github.com/mikegloudemans/gwas-download)
* [h3agwas](https://github.com/h3abionet/h3agwas): GWAS Pipeline for H3Africa





## ATAC
* [ATAC-seq](https://github.com/harvardinformatics/ATAC-seq)
* [ATAC-seq](https://www.encodeproject.org/atac-seq/)
* [atac_dnase_pipelines](https://github.com/kundajelab/atac_dnase_pipelines): ATAC-seq and DNase-seq processing pipeline
* [ATAC-seq Guidelines](https://informatics.fas.harvard.edu/atac-seq-guidelines.html)
* [atac_dnase_pipelines](https://github.com/kundajelab/atac_dnase_pipelines): ATAC-seq and DNase-seq processing pipeline
* [ATAC-seq pipeline](https://www.encodeproject.org/atac-seq/): ATAC-seq Data Standards and Prototype Processing Pipeline

## ChIP
* [ChIPseeker](https://github.com/GuangchuangYu/ChIPseeker): ChIP peak Annotation, Comparison and Visualization
* [metaseq](https://github.com/daler/metaseq): Framework for integrated analysis and plotting of ChIP/RIP/RNA/-seq data
* [pyflow-ChIPseq](https://github.com/crazyhottommy/pyflow-ChIPseq): a snakemake pipeline to process ChIP-seq files from GEO or in-house
* [ChIP-seq-analysis](https://github.com/crazyhottommy/ChIP-seq-analysis): ChIP-seq analysis notes from Tommy Tang
* [chipseq_pipeline](https://github.com/kundajelab/chipseq_pipeline): AQUAS TF and histone ChIP-seq pipeline
* [chip-seq-pipeline2](https://github.com/ENCODE-DCC/chip-seq-pipeline2)

## Methylation
* [Methylation QTL data for brain and blood](https://cnsgenomics.com/software/smr/#DataResource)
* [methylpy](https://github.com/yupenghe/methylpy): WGBS/NOMe-seq Data Processing & Differential 
Methylation Analysis
* [ViewBS](https://github.com/xie186/ViewBS): a powerful toolkit for visualization of high-throughput bisulfite sequencing data
* [mCaller](https://github.com/al-mcintyre/mCaller): A python program to call methylation (m6A in DNA) from nanopore signal data
* [DNA-methylation-analysis](https://github.com/crazyhottommy/DNA-methylation-analysis): notes on DNA methylation analysis (arrays and sequencing data)
* [bs3](https://github.com/khuang28jhu/bs3): BS-Seeker3: An Ultra-fast, Versatile Pipeline for Mapping Bisulfite-treated Reads
* [bsseq](https://github.com/hansenlab/bsseq): Devel repository for bsseq



## Hi-C
* [Hi-C data](http://3dgenome.org)
* [tadtool](https://github.com/vaquerizaslab/tadtool): an interactive tool for the identification of meaningful parameters in TAD-calling algorithms for Hi-C data.
* [juicebox_scripts](https://github.com/phasegenomics/juicebox_scripts): A collection of scripts for working with Hi-C data, Juicebox, and other genomic file formats
* [ALLHiC](https://github.com/tangerzhang/ALLHiC): phasing and scaffolding polyploid genomes based on Hi-C data
* [genomedisco](https://github.com/kundajelab/genomedisco): Software for comparing contact maps from HiC, CaptureC and other 3D genome data
* [3DChromatin_ReplicateQC](https://github.com/kundajelab/3DChromatin_ReplicateQC): Software to compute reproducibility and quality scores for Hi-C data
* [hic_breakfinder](https://github.com/dixonlab/hic_breakfinder)



## 10X data
* [TitanCNA_10X_snakemake](https://github.com/gavinha/TitanCNA_10X_snakemake): Snakemake workflow for 10X Genomics WGS analysis using TitanCNA
* [SVenX](https://github.com/vborjesson/SVenX): Pipeline for SV detection using 10X genomics data
* [10x Genomics](https://github.com/10XGenomics)
* [awesome-10x-genomics](https://github.com/johandahlberg/awesome-10x-genomics): List of tools and resources related to the 10x Genomics GEMCode/Chromium system
* [bxtools](https://github.com/walaj/bxtools): Tools for analyzing 10X Genomics data


## ngs

* [ngsDist](https://github.com/fgvieira/ngsDist)：Estimation of pairwise distances under a probabilistic framework
* [NGS-pipe](https://github.com/cbg-ethz/NGS-pipe): next-generation sequencing pipelines for precision oncology
* [ngsPopGen](https://github.com/mfumagalli/ngsPopGen): Population genetics analyses from NGS data
* [ngsTools](https://github.com/mfumagalli/ngsTools): Programs to analyse NGS data for population genetics purposes
* [viral-ngs](https://github.com/broadinstitute/viral-ngs): Viral genomics analysis pipelines
* [NGSCheckMate](https://github.com/parklab/NGSCheckMate): Software program for checking sample matching for NGS data
* [abtools](https://github.com/briney/abtools): Analysis of antibody NGS data
* [alignment-and-variant-calling-tutorial](https://github.com/ekg/alignment-and-variant-calling-tutorial): basic walk-throughs for alignment and variant calling from NGS sequencing data




## Plotter

* [ClustersPloter](https://github.com/orangeSi/ClustersPloter): visualize genome features cluster
* [Plot chromosome ideograms along with other genomic data](https://gist.github.com/daler/c98fc410282d7570efc3)
* [Laying out multiple plots on a page](https://cran.r-project.org/web/packages/egg/vignettes/Ecosystem.html)
* [plot2DO](https://github.com/rchereji/plot2DO): A tool to assess the quality and distribution of genomic data
* [CMplot](https://github.com/YinLiLin/R-CMplot)
* [ggcyto](https://github.com/RGLab/ggcyto): Visualize Cytometry data with ggplot2
* [ggsashimi](https://github.com/guigolab/ggsashimi): Command-line tool for the visualization of splicing events across multiple samples
* [chromPlot](https://academic.oup.com/bioinformatics/article/32/15/2366/1743488):  visualization of genomic data in chromosomal context
* [karyoploteR](https://bioconductor.org/packages/release/bioc/vignettes/karyoploteR/inst/doc/karyoploteR.html)
* [DataVisualization](https://gist.github.com/KestindotC/7f8caa8be6b33cbe06dbdc5b98d10d6a): Data Visualization in Bioinformatics
* [LocusCompare](http://locuscompare.com): https://github.com/boxiangliu/locuscompare.
* [LocusCompareR](https://github.com/boxiangliu/locuscomparer)


## UMI
* [zUMIs](https://github.com/sdparekh/zUMIs): A fast and flexible pipeline to process RNA sequencing data with UMIs
* [umis](https://github.com/vals/umis): Tools for processing UMI RNA-tag data


## TCR
* [tcR](https://github.com/cran/tcR): Advanced Data Analysis of Immune Receptor Repertoires



## Bioinformatics tutorial

* [NovoAlign NGS Quick Start Tutorial](http://www.novocraft.com/documentation/novoalign-2/novoalign-ngs-quick-start-tutorial/)
* [In-depth-NGS-Data-Analysis-Course](https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course): Materials for 12-day course on analyzing RNA-Seq, ChIP-Seq and variant calling data https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/
* [LncPipe](https://github.com/likelet/LncPipe): A Nextflow-based pipeline for comprehensive analyses of long non-coding RNAs from RNA-seq datasets
* [NGI-ChIPseq](https://github.com/SciLifeLab/NGI-ChIPseq): Nextflow ChIP-seq data analysis pipeline, National Genomics Infrastructure, Science for Life Laboratory in Stockholm
* [nf-GATK_Exome_Preprocess](https://github.com/joshua-d-campbell/nf-GATK_Exome_Preprocess): Adapted from the GATK best practice guide to preprocess whole exome sequencing (WES) data
* [pyflow-ATACseq](https://github.com/crazyhottommy/pyflow-ATACseq): ATAC-seq snakemake pipeline
* [pyflow-ChIPseq](https://github.com/crazyhottommy/pyflow-ChIPseq): a snakemake pipeline to process ChIP-seq files from GEO or in-house
* [single-cell-rna-seq](https://github.com/snakemake-workflows/single-cell-rna-seq): A single cell RNA-seq workflow following http://dx.doi.org/10.12688/f1000research.9501.2
* [pyflow-RNAseq](https://github.com/crazyhottommy/pyflow-RNAseq): RNAseq pipeline based on snakemake
* [atac-seq-pipeline](https://github.com/ENCODE-DCC/atac-seq-pipeline): ENCODE ATAC-seq pipeline
* [atac_dnase_pipelines](https://github.com/kundajelab/atac_dnase_pipelines): ATAC-seq and DNase-seq processing pipeline
* [m6A-seq_analysis_workflow](https://github.com/scottzijiezhang/m6A-seq_analysis_workflow): A pipeline to process m6A-seq data and down stream analysis
* [Harvard FAS Tutorials](https://informatics.fas.harvard.edu/category/tutorials.html): Harvard FAS Tutorials and Training
* [ENCODE ATAC-seq](https://www.encodeproject.org/search/?type=Pipeline&assay_term_names=ATAC-seq): ATAC-seq pipeline
* [getting-started-with-genomics-tools-and-resources](https://github.com/crazyhottommy/getting-started-with-genomics-tools-and-resources): Unix, R and python tools for genomics
* [DNA-seq-analysis](https://github.com/crazyhottommy/DNA-seq-analysis): notes on whole exome and whole genome sequencing analysis
* [RNA-seq-analysis](https://github.com/crazyhottommy/RNA-seq-analysis): RNAseq analysis notes from Tommy Tang
* [scRNAseq-analysis-notes](https://github.com/crazyhottommy/scRNAseq-analysis-notes): scRNAseq analysis notes
* [Zika-RNAseq-Pipeline](https://github.com/MaayanLab/Zika-RNAseq-Pipeline): An open RNA-Seq data analysis pipeline tutorial with an example of reprocessing data from a recent Zika virus study
* [superFreq](https://github.com/ChristofferFlensburg/superFreq): Analysis pipeline for cancer exomes



## database & Websites

* [Segmental duplications (hg38)](https://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/genomicSuperDups.txt.gz)
* [self-chains (hg38)](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/chainSelf.txt.gz)
* [Pfam domains (hg38)](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/ucscGenePfam.txt.gz)
* [hg38 gtf](https://ftp.ensembl.org/pub/release-96/gtf/homo_sapiens/Homo_sapiens.GRCh38.96.gtf.gz)
* [gnomAD](https://storage.googleapis.com/gnomad-public/release/2.0.1/vcf/exomes/gnomad.exomes.r2.0.1.sites.vcf.gz)
* [CADD files for both indels and SNPs (hg38)](http://krishna.gs.washington.edu/download/CADD/v1.5/GRCh38/)
* [MPC](https://ftp.broadinstitute.org/pub/ExAC_release/release1/regional_missense_constraint/fordist_constraint_official_mpc_values.txt.gz)
* [whole-exome MTR file](http://mtr-viewer.mdhs.unimelb.edu.au:8079/mtrflatfile_1.0.txt.gz)
* [REVEL file](https://rothsj06.u.hpc.mssm.edu/revel/revel_all_chromosomes.csv.zip)
* [pLI file](https://ftp.broadinstitute.org/pub/ExAC_release/release1/manuscript_data/forweb_cleaned_exac_r03_march16_z_data_pLI.txt.gz)
* [ClinVar VCF file](https://ftp.ncbi.nih.gov/pub/clinvar/vcf_GRCh37/archive_2.0/2017/clinvar_20170802.vcf.gz)
* [de novo variants on s3 server](https://s3.us-east-2.amazonaws.com/pathoscore-data/samocha/samochadenovo.xlsx)
* [Short Tandem Repeat DNA](https://strbase.nist.gov/)
* [OMIM](http://www.omim.org/)
* [LDlink](https://ldlink.nci.nih.gov/)
* [SIFT](http://sift.jcvi.org/)
* [Polyphen-2](http://genetics.bwh.harvard.edu/pph2/)
* [MutationTaster](http://www.mutationtaster.org/)
* [ESP6500](http://evs.gs.washington.edu/EVS/)
* [ExAC](http://exac.broadinstitute.org/)
* [HGMD](http://www.biobase-international.com/product/)
* [Tohoku Medical Megabank Genome Reference Panel](https://jmorp.megabank.tohoku.ac.jp/)
* [iJGVD](https://ijgvd.megabank.tohoku.ac.jp/)
* [UK Biobank](http://ukbiobank.ac.uk)
* [Database of Genotypes and Phenotypes (dbGaP)]https://www.ncbi.nlm.nih.gov/gap
* [FUMA software](http://fuma.ctglab.nl)
* [MAGMA software](http://ctg.cncr.nl/software/magma)
* [mvGWAMA and effective sample size calculation](https://github.com/Kyoko-wtnb/mvGWAMA)
* [LD Score Regression software](https://github.com/bulik/ldsc)
* [LD Hub (GWAS summary statistics)](http://ldsc.broadinstitute.org/)
* [LD scores](https://data.broadinstitute.org/alkesgroup/LDSCORE/)
* [Psychiatric Genomics Consortium (GWAS summary statistics)](http://www.med.unc.edu/pgc/results-and-downloads)
* [MSigDB curated gene-set database](http://software.broadinstitute.org/gsea/)
* [msigdb/collections.jsp; NHGRI GWAS catalog](https://www.ebi.ac.uk/gwas/)
* [GSMR software](http://cnsgenomics.com/software/gsmr/)
* [credible SNP set analysis software](https://github.com/hailianghuang/FM-summary)
* [IGSR: The International Genome Sample Resource](http://www.internationalgenome.org/data#download): Providing ongoing support for the 1000 Genomes Project data
* [1000genome vcf](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/GRCh38_positions/)
* [decipher](https://decipher.sanger.ac.uk/)
* [DGVa](https://www.ebi.ac.uk/dgva): Database of Genomic Variants archive
* [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/)
* [COSMIC](https://cancer.sanger.ac.uk/cosmic)
* [Database of Genomic Variants](http://dgv.tcag.ca/dgv/app/home)
* [ENCODE](http://encodeproject.org/)
* [3D Genome Browser](http://3dgenome.org/)
* [HPO: The Human Phenotype Ontology](https://hpo.jax.org/app/)
* [CHPO](http://www.chinahpo.org/)
* [ICGC](https://icgc.org/): International Cancer Genome Consortium
* [GWAS Catalog](https://www.ebi.ac.uk/gwas/): The NHGRI-EBI Catalog of published genome-wide association studies
* [RefSeq GRCh38](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/refGene.txt.gz)
* [Deciphering Developmental Disorders (DDD) Study](https://www.ddduk.org/)
* [OMIM (Online Mendelian Inheritance in Man)](https://www.omim.org/): https://omim.org/downloads/
* [gnomAD](https://gnomad.broadinstitute.org/)
* [dbVar](https://www.ncbi.nlm.nih.gov/dbvar/content/human_hub/): Non-Redundant Structural Variation Datasets
* [GeneHancer](https://www.genecards.org/)
* [Exome Aggregation Consortium (ExAC)](http://exac.broadinstitute.org/)
* [dbVar](https://www.ncbi.nlm.nih.gov/dbvar)
* [American College of Medical Genetics and Genomics (ACMG)](https://www.acmg.net/)
* [VarCards](http://varcards.biols.ac.cn/): an integrated genetic and clinical database for coding variants in the human genome
* [1000 Genomes Consortium GRCh38](http://www.internationalgenome.org/data-portal/search?q=GRCh38)
* [Replication Domain genome browser and analysis tool](www.replicationdomain.com)
* [European Nucleotide Archive](https://www.ebi.ac.uk/ena)
* [Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra)
* [ANNOVAR](http://annovar.openbioinformatics.org/)
* [CLINVITAE](http://clinvitae.invitae.com/)
* [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP)
* [dbSNP](http://www.ncbi.nlm.nih.gov/SNP)
* [Ensembl]() http://www.ensembl.org/)
* [GERPþþ](http://mendel.stanford.edu/SidowLab/downloads/gerp/)
* [GWASdb](http://jjwanglab.org/gwasdb)
* [HGMD](http://www.hgmd.org)
* [InterVar](https://github.com/WGLab/InterVar)
* [MedGen](https://www.ncbi.nlm.nih.gov/medgen/)
* [NHLBI Exome Sequencing Project (ESP) Exome Variant Server](http://evs.gs.washington.edu/EVS/)
* [OMIM](http://omim.org/)
* [OrphaNet](http://www.orpha.net/)
* [PolyPhen-2](http://genetics.bwh.harvard.edu/pph2)
* [RefSeq](http://www.ncbi.nlm.nih.gov/refseq)
* [RepeatMasker](http://www.repeatmasker.org/)
* [SIFT](http://sift.jcvi.org/)
* [UCSC Genome Browser](http://genome.ucsc.edu)
* [wIntervar](http://wintervar.wglab.org/)
* [GenomeNet](http://www.genome.jp/): integrated database 
* [KEGG](http://www.genome.jp/kegg/): Kyoto Encyclopedia of Genes and Genomes
* [KOBAS](http://kobas.cbi.pku.edu.cn/index.php)
* [InterPro](https://www.ebi.ac.uk/interpro/download.html)
* [ENCODE](https://www.encodeproject.org/): ENCODE
* [HGNC](https://www.genenames.org): is responsible for approving unique symbols and names for human loci, including protein coding genes, ncRNA genes and pseudogenes, to allow unambiguous scientific communication.
* [InterVar](https://github.com/WGLab/InterVar): A bioinformatics software tool for clinical interpretation of genetic variants by the 2015 ACMG-AMP guideline
* [Ira Hall lab](https://github.com/hall-lab)
* [IPD-IMGT/HLA](http://www.ebi.ac.uk/ipd/imgt/hla/): https://github.com/ANHIG/IMGTHLA
* [iPSYCH download site](http://ipsych.au.dk/downloads/)
* [iPSYCH project](http://ipsych.au.dk/)
* [LISA cluster at SURFsara](https://userinfo.surfsara.nl/systems/lisa/)
* [Lisa Genetic Cluster Computer](http://geneticcluster.org/)
* [/giab/ftp/data/NA12878](https://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/data/NA12878/)
* [GTEx eQTL data](https://gtexportal.org/home/datasets)
* [GTEx portal](https://gtexportal.org/home/)
* [International Paediatric and Congenital Cardiac Codes](http://www.ipccc.net/)
* [InterProScan](http://www.ebi.ac.uk/interpro/download.html)
* [genome-in-a-bottle](https://github.com/genome-in-a-bottle): A public-private-academic consortium hosted by NIST to develop reference materials and standards for clinical sequencing


## Deal with vcf
* [cyvcf2](https://github.com/brentp/cyvcf2): fast VCF and BCF processing	
* [CyVCF document](http://pyvcf.readthedocs.io/en/latest/)
* [CyVCF](https://github.com/arq5x/cyvcf): A fast Python library for VCF files leveraging Cython for speed.
* [rtg-tools](https://github.com/RealTimeGenomics/rtg-tools): Utilities for accurate VCF comparison and manipulation
* [spVCF](https://github.com/mlin/spVCF): Sparse Project VCF: evolution of VCF to encode population genotype matrices efficiently
* [vcf2phylip](https://github.com/edgardomortiz/vcf2phylip): Convert SNPs in VCF format to PHYLIP, NEXUS, binary NEXUS, or FASTA alignments for phylogenetic analysis
* [vcflib](https://github.com/vcflib/vcflib): a simple C++ library for parsing and manipulating VCF files, + many command-line utilities






## Websites for cancer data

- [The Cancer Genome Atlas (TCGA)](https://portal.gdc.cancer.gov/)
- [International Cancer Genome Consortium (ICGC)](https://icgc.org/)
- [Catalogue of Somatic Mutations in Cancer (COSMIC)](https://cancer.sanger.ac.uk/cosmic/)
- [cBioPortal](http://www.cbioportal.org/):  provides visualization, analysis and download of large-scale cancer genomics data sets
- [Pan-Cancer Atlas](https://www.cell.com/pb-assets/consortium/pancanceratlas/pancani3/index.html#group-oncogenic-processes-1HqWoUt3SE)
- [DoCM](http://www.docm.info/): database of curated mutations
- [CKB](https://ckb.jax.org/gene/grid): JAX-Clinical Knowledgebase
- [OncoKB](http://oncokb.org/#/cancerGenes): OncoKB Cancer Genes
- [PMKB](https://pmkb.weill.cornell.edu/): Precision Medicine Knowledgebase
- [GeneCards](https://www.genecards.org/cgi-bin/carddisp.pl?gene=TP53): human gene database
- [MalaCards](http://www.malacards.org): The human disease database
* [Cancer Cell Line Encyclopedia](https://portals.broadinstitute.org/ccle)
* [MiOncoCirc](https://mioncocirc.github.io): A compendium of circular RNAs compiled from cancer clinical samples at The University of Michigan


## Blogs

* [genesino](http://blog.genesino.com/): 生信宝典
* [qinqianshan](http://qinqianshan.com)
* [nameix](https://blog.csdn.net/nameix)
* [biotrainee](https://vip.biotrainee.com/)
* [Biostack](http://www.biostack.org/)
* [chenlianfu](http://www.chenlianfu.com/)


## Labs
* [Trapnell Lab](http://cole-trapnell-lab.github.io/): aim to identify genes that control cellular transitions, primarily using single-cell genomics
* [H3ABioNet](https://github.com/h3abionet)
* [Pacific Biosciences](https://github.com/PacificBiosciences)


## Tool resources

* [SciHub desktop](https://zhuanlan.zhihu.com/p/31809890)
* [sci-hub](http://sci-hub.hk/): http://sci-hub.tw/, http://sci-hub.hk/, http://www.sci-hub.cn/
* [libgen](http://gen.lib.rus.ec/)
* [search code](https://searchcode.com/)
* [omictools](https://omictools.com/)
* [anaconda cloud](https://anaconda.org)
* [Gene Regulation Info](http://generegulation.info/index.php/dna-protein-binding): Protein-DNA binding: data, tools & models
* [STHDA](http://www.sthda.com/french/): Statistical tools for high-throughput data analysis
* [awesome-cancer-variant-databases](https://github.com/seandavi/awesome-cancer-variant-databases): A community-maintained repository of cancer clinical knowledge bases and databases focused on cancer variants.




## Python
* [libpku](https://github.com/lib-pku/libpku): 贵校课程资料民间整理 (https://lib-pku.github.io/)
* [REKCARC-TSC-UHT](https://github.com/PKUanonym/REKCARC-TSC-UHT): 清华大学计算机系课程攻略
* [USTC-Course](https://github.com/USTC-Resource/USTC-Course): 中国科学技术大学课程资源 (https://ustc-resource.github.io/USTC-Course/)
* [linux-command](https://github.com/jaywcjlove/linux-command): Linux命令大全搜索工具，内容包含Linux命令手册、详解、学习、搜集。https://git.io/linux https://git.io/linux
* [Algorithm_Interview_Notes-Chinese](https://github.com/imhuay/Algorithm_Interview_Notes-Chinese)
* [awesome-programming-books](https://github.com/majikarp/awesome-programming-books): A curated list of awesome programming books
* [TheAlgorithms/Python](https://github.com/TheAlgorithms/Python): All Algorithms implemented in Python
* [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-Plus): All Algorithms implemented in C++
* [git-tips](https://github.com/521xueweihan/git-tips)
* [PySnooper](https://github.com/cool-RR/PySnooper): Never use print for debugging again (a poor man debugger)
* [pandas-cookbook](https://github.com/jvns/pandas-cookbook): Recipes for using Python pandas library



## Machine Learning

* [tensorflow_cookbook](https://github.com/nndl/tensorflow_cookbook): Code for Tensorflow Machine Learning Cookbook
* [pumpkin-book](https://github.com/datawhalechina/pumpkin-book): 机器学习》（西瓜书）公式推导解析，在线阅读地址：https://datawhalechina.github.io/pumpkin-book
* [deeplearningbook-chinese](https://github.com/exacity/deeplearningbook-chinese)
* [nndl.github.io](https://github.com/nndl/nndl.github.io): 《神经网络与深度学习》 Neural Network and Deep Learning https://nndl.github.io
* [DeepLearning_Summary](https://github.com/zhwhong/DeepLearning_Summary): A list of awesome Deep Learning tutorials, projects and communities. Forked from other github repositories.
* [Paddle](https://github.com/PaddlePaddle/Paddle): PArallel Distributed Deep LEarning
* [parallel_ml_tutorial](https://github.com/ogrisel/parallel_ml_tutorial): Tutorial on scikit-learn and IPython for parallel machine learning
* [DragoNN](https://kundajelab.github.io/dragonn/tutorials.html)
* [Kaggle machine learning competitions](https://www.kaggle.com/sudalairajkumar/winning-solutions-of-kaggle-competitions)
* [Keras](https://keras.io/)
* [Keras model zoos](https://keras.io/applications/)
* [PyTorch](http://pytorch.org)
* [PyTorch model zoos](https://pytorch.org/docs/stable/torchvision/models.html)
* [Tensorflow model zoos](https://github.com/tensorflow/models) 
* [tensorflow](https://www.tensorflow.org/hub/)
* [STNeuroNet](https://github.com/soltanianzadeh/STNeuroNet): Software for the paper "Fast and robust active neuron segmentation in two-photon calcium imaging using spatio-temporal deep learning," Proceedings of the National Academy of Sciences (PNAS), 2019
* [deep-q-learning](https://github.com/keon/deep-q-learning): Minimal Deep Q Learning (DQN & DDQN) implementations in Keras
* [DeepSVR](https://github.com/griffithlab/DeepSVR): a machine learning model approach to somatic variant refinement
* [DeepLearning-500-questions](https://github.com/scutan90/DeepLearning-500-questions): 深度学习500问
* [DeepSVR](https://github.com/griffithlab/DeepSVR): a machine learning model approach to somatic variant refinement
* [deepvariant](https://github.com/google/deepvariant): an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA sequencing data
* [deTiN](https://github.com/broadinstitute/deTiN): DeTiN is designed to measure tumor-in-normal contamination and improve somatic variant detection sensitivity when using a contaminated matched control.
* [homemade-machine-learning](https://github.com/trekhleb/homemade-machine-learning): Python examples of popular machine learning algorithms with interactive Jupyter demos and math being explained
* [practicalAI](https://github.com/GokuMohandas/practicalAI): A practical approach to learning machine learning
* [vowpal_wabbit](https://github.com/JohnLangford/vowpal_wabbit):  a machine learning system which pushes the frontier of machine learning with techniques such as online, hashing, allreduce, reductions, learning2search, active, and interactive learning.
* [janggu](https://github.com/BIMSBbioinfo/janggu): Deep learning infrastructure for bioinformatics
* [learn-python](https://github.com/trekhleb/learn-python): Playground and cheatsheet for learning Python. Collection of Python scripts that are split by topics and contain code examples with explanations.
* [TensorFlow-Course](https://github.com/open-source-for-science/TensorFlow-Course)
* [awesome-nlp](https://github.com/keon/awesome-nlp): A curated list of resources dedicated to Natural Language Processing (NLP)
* [Coursera-ML-AndrewNg-Notes](https://github.com/fengdu78/Coursera-ML-AndrewNg-Notes): 吴恩达老师的机器学习课程个人笔记
* [flair](https://github.com/zalandoresearch/flair): A very simple framework for state-of-the-art Natural Language Processing (NLP)





