# Structural variantion


## SV Caller for third generation sequences
* [svim](https://github.com/eldariont/svim): Structural Variant Identification Method using Long Reads
* [SURVIVOR](https://github.com/fritzsedlazeck/SURVIVOR): Toolset for SV simulation, comparison and filtering
* [Sniffles](https://github.com/fritzsedlazeck/Sniffles): Structural variation caller using third generation sequencing
* [NanoSV](https://github.com/mroosmalen/nanosv): SV caller for nanopore data


## SV on illumina data
* [svtyper](https://github.com/hall-lab/svtyper): Bayesian genotyper for structural variants
* [lumpy-sv](https://github.com/arq5x/lumpy-sv): a general probabilistic framework for structural variant discovery
* [parliament2](https://github.com/dnanexus/parliament2): Runs a combination of tools to generate structural variant calls on whole-genome sequencing data
* [delly](https://github.com/dellytools/delly): Structural variant discovery by integrated paired-end and split-read analysis
* [manta](https://github.com/Illumina/manta): Structural variant and indel caller for mapped sequencing data
* [parliament2](https://github.com/dnanexus/parliament2): Runs a combination of tools to generate structural variant calls on whole-genome sequencing data
* [SV2](https://github.com/dantaki/SV2): Support Vector Structural Variation Genotyper
* [pindel](https://github.com/genome/pindel):  identify the breakpoints of these variants from paired-end short reads
* [MetaSV](https://bioinform.github.io/metasv/): An accurate and integrative structural-variant caller for next generation sequencing
* [svaba](https://github.com/walaj/svaba): Structural variation and indel detection by local assembly
* [wham](https://github.com/zeeev/wham): Structural variant detection and association testing
* [gridss](https://github.com/PapenfussLab/gridss): Genomic Rearrangement IDentification Software Suite

## CNV

* [Control-FREE](http://boevalab.com/FREEC/): a tool for assessing copy number and allelic content using next generation sequencing data



## General
* [svaba](https://github.com/walaj/svaba): Structural variation and indel detection by local assembly
* [truvari](https://github.com/spiralgenetics/truvari): Structural variant comparison tool for VCFs
* [smoove](https://github.com/brentp/smoove): structural variant calling and genotyping with existing tools, but, smoothly
