# Bioinformatics tutorial

* [NovoAlign NGS Quick Start Tutorial](http://www.novocraft.com/documentation/novoalign-2/novoalign-ngs-quick-start-tutorial/)
* [In-depth-NGS-Data-Analysis-Course](https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course): Materials for 12-day course on analyzing RNA-Seq, ChIP-Seq and variant calling data https://hbctraining.github.io/In-depth-NGS-Data-Analysis-Course/


# Pipelines

* [LncPipe](https://github.com/likelet/LncPipe): A Nextflow-based pipeline for comprehensive analyses of long non-coding RNAs from RNA-seq datasets
* [NGI-ChIPseq](https://github.com/SciLifeLab/NGI-ChIPseq): Nextflow ChIP-seq data analysis pipeline, National Genomics Infrastructure, Science for Life Laboratory in Stockholm
* [nf-GATK_Exome_Preprocess](https://github.com/joshua-d-campbell/nf-GATK_Exome_Preprocess): Adapted from the GATK best practice guide to preprocess whole exome sequencing (WES) data
* [pyflow-ATACseq](https://github.com/crazyhottommy/pyflow-ATACseq): ATAC-seq snakemake pipeline
* [pyflow-ChIPseq](https://github.com/crazyhottommy/pyflow-ChIPseq): a snakemake pipeline to process ChIP-seq files from GEO or in-house
* [single-cell-rna-seq](https://github.com/snakemake-workflows/single-cell-rna-seq): A single cell RNA-seq workflow following http://dx.doi.org/10.12688/f1000research.9501.2
* [pyflow-RNAseq](https://github.com/crazyhottommy/pyflow-RNAseq): RNAseq pipeline based on snakemake
* [atac-seq-pipeline](https://github.com/ENCODE-DCC/atac-seq-pipeline): ENCODE ATAC-seq pipeline
* [atac_dnase_pipelines](https://github.com/kundajelab/atac_dnase_pipelines): ATAC-seq and DNase-seq processing pipeline
* [m6A-seq_analysis_workflow](https://github.com/scottzijiezhang/m6A-seq_analysis_workflow): A pipeline to process m6A-seq data and down stream analysis


# Blogs

* [genesino](http://blog.genesino.com/): 生信宝典
* [qinqianshan](http://qinqianshan.com)
* [nameix](https://blog.csdn.net/nameix)

# Resources

* [getting-started-with-genomics-tools-and-resources](https://github.com/crazyhottommy/getting-started-with-genomics-tools-and-resources): Unix, R and python tools for genomics
* [DNA-seq-analysis](https://github.com/crazyhottommy/DNA-seq-analysis): notes on whole exome and whole genome sequencing analysis
* [RNA-seq-analysis](https://github.com/crazyhottommy/RNA-seq-analysis): RNAseq analysis notes from Tommy Tang
* [scRNAseq-analysis-notes](https://github.com/crazyhottommy/scRNAseq-analysis-notes): scRNAseq analysis notes
* [DNA-methylation-analysis](https://github.com/crazyhottommy/DNA-methylation-analysis): notes on DNA methylation analysis (arrays and sequencing data)
* [pandas-cookbook](https://github.com/jvns/pandas-cookbook): Recipes for using Python's pandas library
* [NucleoATAC](https://github.com/GreenleafLab/NucleoATAC): nucleosome calling using ATAC-seq
* [mCaller](https://github.com/al-mcintyre/mCaller): A python program to call methylation (m6A in DNA) from nanopore signal data
