Chr
Start
End
Ref
Alt
Func.ensGene_dechr
Gene.ensGene_dechr
GeneDetail.ensGene_dechr
ExonicFunc.ensGene_dechr
AAChange.ensGene_dechr
cytoband
avsnp150: dbSNP ID
ExAC_ALL: All
ExAC_AFR: African
ExAC_AMR: Admixed American
ExAC_EAS: East Asian
ExAC_FIN: Finnish
ExAC_NFE: Non-Finnish European
ExAC_OTH: Other
ExAC_SAS: South Asian
ExAC_nonpsych_ALL
ExAC_nonpsych_AFR
ExAC_nonpsych_AMR
ExAC_nonpsych_EAS
ExAC_nonpsych_FIN
ExAC_nonpsych_NFE
ExAC_nonpsych_OTH
ExAC_nonpsych_SAS
gnomAD_exome_ALL
gnomAD_exome_AFR
gnomAD_exome_AMR
gnomAD_exome_ASJ
gnomAD_exome_EAS
gnomAD_exome_FIN
gnomAD_exome_NFE
gnomAD_exome_OTH
gnomAD_exome_SAS
gnomAD_genome_ALL
gnomAD_genome_AFR
gnomAD_genome_AMR
gnomAD_genome_ASJ
gnomAD_genome_EAS
gnomAD_genome_FIN
gnomAD_genome_NFE
gnomAD_genome_OTH
esp6500siv2_all: All 6503 Individuals exomes
GME_AF: Great Middle East 项目人群中频率人群分类为NWA (northwest Africa), NEA (northeast Africa), AP (Arabian peninsula), Israel, SD (Syrian desert), TP (Turkish peninsula) and CA (Central Asia)
GME_NWA
GME_NEA
GME_AP
GME_Israel
GME_SD
GME_TP
GME_CA
HRC_AF
HRC_AC
HRC_AN
HRC_non1000G_AF
HRC_non1000G_AC
HRC_non1000G_AN
Kaviar_AF: Kaviar contains 162 million SNV sites (including 25M not in dbSNP) and incorporates data from 35 projects encompassing 77,781 individuals (13.2K whole genome, 64.6K exome), Alternative frequency
Kaviar_AC: Alternative count
Kaviar_AN: Total number
cosmic70: COSMIC database
CLNALLELEID
CLNDN
CLNDISDB
CLNREVSTAT
CLNSIG: SIG refers to Variant Clinical Significance, including unknown, untested, non-pathogenic, probable-non-pathogenic, probable-pathogenic, pathogenic, drug-response,histocompatibility, other



SIFT_score: SIFT score (SIFTori). Scores range from 0 to 1. The smaller the score the more likely the SNP has damaging effect
SIFT_converted_rankscore: SIFTori scores were first converted to SIFTnew=1-SIFTori, then ranked among all SIFTnew scores in dbNSFP. The rankscore is the ratio of  the rank the SIFTnew score over the total number of SIFTnew scores in dbNSFP. If there are multiple scores, only the most damaging (largest) rankscore is presented. The rankscores range from 0.02654 to 0.87932
SIFT_pred:  SIFT prediction, D: Deleterious (sift<=0.05); T: tolerated (sift>0.05)
Polyphen2_HDIV_score: Pholyphen2 score based on HDIV. The score ranges from 0 to 1
Polyphen2_HDIV_rankscore: Polyphen2 HDIV scores were first ranked among all HDIV scores in dbNSFP. The rankscore is the ratio of the rank the score over the total number of the scores in dbNSFP. If there are multiple scores, only the most damaging (largest) rankscore is presented. The scores range from 0.02656 to 0.89917
Polyphen2_HDIV_pred: Pholyphen2 prediction based on HDIV, D: Probably damaging (>=0.957), P: possibly damaging (0.453<=pp2_hdiv<=0.956); B: benign (pp2_hdiv<=0.452)
Polyphen2_HVAR_score: Polyphen2 score based on HVAR
Polyphen2_HVAR_rankscore: Polyphen2 HVAR scores were first ranked among all HVAR scores in dbNSFP. The rankscore is the ratio of the rank the score over the total number of the scores in dbNSFP. If there are multiple scores, only the most damaging (largest) rankscore is presented. The scores range from 0.01281 to 0.9711
Polyphen2_HVAR_pred: Polyphen2 prediction based on HVAR, D: Probably damaging (>=0.909), P: possibly damaging (0.447<=pp2_hdiv<=0.909); B: benign (pp2_hdiv<=0.446)
LRT_score: The original LRT two-sided p-value (LRTori), ranges from 0 to 1
LRT_converted_rankscore: LRTori scores were first converted as LRTnew=1-LRTori*0.5 if Omega<1, or LRTnew=LRTori*0.5 if Omega>=1. Then LRTnew scores were ranked among all LRTnew scores in dbNSFP. The rankscore is the ratio of the rank over the total number of the scores in dbNSFP. The scores range from 0.00166 to 0.85682.
LRT_pred: LRT prediction, D: Deleterious; N: Neutral; U: Unknown, Lower scores are more deleterious
MutationTaster_score: MutationTaster p-value (MTori), ranges from 0 to 1
MutationTaster_converted_rankscore: The MTori scores were first converted. If the prediction is "A" or "D" MTnew=MTori; if the prediction is "N" or "P", MTnew=1-MTori. Then MTnew 	scores were ranked among all MTnew scores in dbNSFP. The rankscore is the ratio of the	rank of the score over the total number of MTnew scores in dbNSFP. The scores range	from 0.09067 to 0.80722.
MutationTaster_pred: MutationTaster prediction, A" ("disease_causing_automatic"); "D" ("disease_causing"); "N" ("polymorphism"); "P" ("polymorphism_automatic"). The score cutoff between "D" and "N" is 0.5 for MTori and 0.31655 for the rankscore.
MutationAssessor_score: MutationAssessor functional impact combined score (MAori). The score ranges from -5.135 to 6.49 in dbNSFP
MutationAssessor_score_rankscore: MAori scores were ranked among all MAori scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of MAori scores in dbNSFP. The scores range from 0 to 1
MutationAssessor_pred: MutationAssessor prediction, H: high; M: medium; L: low; N: neutral. H/M means functional and L/N means non-functional. The MAori score cutoffs between "H" and "M", "M" and "L", and "L" and "N", are 3.5, 1.935 and 0.8, respectively. The rankscore cutoffs between "H" and "M", "M" and "L", and "L" and "N", are 0.92924, 0.51945 and 0.19692, respectively
FATHMM_score: FATHMM default score (weighted for human inherited-disease mutations with	Disease Ontology) (FATHMMori). Scores range from -18.09 to 11.0. The smaller the score the more likely the SNP has damaging effect
FATHMM_converted_rankscore: FATHMMnew=-FATHMMori were ranked among all FATHMMnew scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of FATHMMnew scores in dbNSFP. If there are multiple scores, only the most damaging (smallest) rankscore is presented. The scores range from 0 to 1.
FATHMM_pred: FATHMM prediction, D: Deleterious; T: Tolerated. If a FATHMMori score is <=-1.5 (or rankscore >=0.81415) the corresponding NS is predicted as "D(AMAGING)"; otherwise it is predicted as "T(OLERATED)"
PROVEAN_score: PROVEAN score (PROVEANori). Scores range from -14 to 14. The smaller the score the more likely the SNP has damaging effect
PROVEAN_converted_rankscore: PROVEANori were first converted to PROVEANnew=1-(PROVEANori+14)/28, then ranked among all PROVEANnew scores in dbNSFP. The rankscore is the ratio of the rank the PROVEANnew score over the total number of PROVEANnew scores in dbNSFP. If there are multiple scores, only the most damaging (largest) rankscore is presented.
PROVEAN_pred: PROVEAN prediction, D: Deleterious;  N: Neutral, higher values are more deleterious
VEST3_score: VEST 3.0 score. Score ranges from 0 to 1. The larger the score the more likely	the mutation may cause functional change. In case there are multiple scores for the same variant, the largest score (most damaging) is presented.
VEST3_rankscore: VEST3 scores were ranked among all VEST3 scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of VEST3 	scores in dbNSFP. The scores range from 0 to 1
MetaSVM_score: Our support vector machine (SVM) based ensemble prediction score, which incorporated 10 scores (SIFT, PolyPhen-2 HDIV, PolyPhen-2 HVAR, GERP++, MutationTaster, 	Mutation Assessor, FATHMM, LRT, SiPhy, PhyloP) and the maximum frequency observed in the 1000 genomes populations. Larger value means the SNV is more likely to be damaging. Scores range from -2 to 3 in dbNSFP
MetaSVM_rankscore: MetaSVM scores were ranked among all MetaSVM scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of MetaSVM 	scores in dbNSFP. The scores range from 0 to 1
MetaSVM_pred: MetaSVM prediction, D: Deleterious; T: Tolerated, The rankscore cutoff between "D" and "T" is 0.83357, higher scores are more deleterious
MetaLR_score: Our logistic regression (LR) based ensemble prediction score, which incorporated 10 scores (SIFT, PolyPhen-2 HDIV, PolyPhen-2 HVAR, GERP++, MutationTaster, Mutation Assessor, FATHMM, LRT, SiPhy, PhyloP) and the maximum frequency observed in the 1000 genomes populations. Larger value means the SNV is more likely to be damaging. Scores range from 0 to 1.
MetaLR_rankscore: MetaLR scores were ranked among all MetaLR scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of MetaLR scores in dbNSFP. The scores range from 0 to 1.
MetaLR_pred: D: Deleterious; T: Tolerated, higher scores are more deleterious. The score cutoff between "D" and "T" is 0.5. The rankscore cutoff between "D" and "T" is 0.82268.
M-CAP_score: M-CAP score
M-CAP_rankscore: M-CAP scores were ranked among all M-CAP scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of M-CAP scores in dbNSFP
M-CAP_pred: M-CAP prediction, The score cutoff between "D(amaging)" and "T(olerated)" is 0.025, 
REVEL_score: REVEL is an ensemble score based on 13 individual scores for predicting the pathogenicity of missense variants. Scores range from 0 to 1. The larger the score the more likely the SNP has damaging effect
REVEL_rankscore: REVEL scores were ranked among all REVEL scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of REVEL scores in dbNSFP
MutPred_score: General MutPred score. Scores range from 0 to 1. The larger the score the more likely the SNP has damaging effect
MutPred_rankscore: MutPred scores were ranked among all MutPred scores in dbNSFP. The rankscore is the ratio of the rank of the score over the total number of MutPred scores in dbNSFP.
CADD_raw: CADD (Combined Annotation Dependent Depletion) is a score that is based on SVM on multiple other scores.  Raw values have relative meaning, with higher values indicating that a variant is more likely to have deleterious effects.
CADD_raw_rankscore: the rank of the CADD_raw score among all CADD_raw scores in genome
CADD_phred: CADD phred-like score, ranges 1-99, the larger the number the more likely damaging; score >10 means the variant in the top 10% (0.1) among the total 8.6 billion possible SNVs, >20 means in the top 1%, >30 means in the top 0.1%, etc. CADD suggests a cutoff between 10 and 20 (e.g. 15)
DANN_rankscore: rank of the DANN_score among all DANN_scores
fathmm-MKL_coding_score: 
fathmm-MKL_coding_rankscore: 
fathmm-MKL_coding_pred: 
Eigen_coding_or_noncoding: Whether the Eigen and EigenPC scores are based on coding model or noncoding model
Eigen-raw: Eigen score for coding SNVs. A functional prediction score based on conservation, allele frequencies, and deleteriousness prediction using an unsupervised learning method
Eigen-PC-raw: Eigen PC score for genome-wide SNVs. A functional prediction score based on conservation, allele frequencies, deleteriousness prediction (for missense SNVs) and epigenomic signals (for synonymous and non-coding SNVs) using an unsupervised learning method (doi: 10.1038/ng.3477)
GenoCanyon_score: a functional prediction score based on conservation and biochemical annotations using unsupervised statistical learning
GenoCanyon_score_rankscore: rank of the GenoCanyon_score among all GenoCanyon_scores in genome
integrated_fitCons_score: fitCons scores (i6) based on function evidence from multiple cell types, the higher the score the more potential for interesting genomic function
integrated_fitCons_score_rankscore: rank of the integrated_fitCons_score among all integrated_fitCons_scores in genome
integrated_confidence_value: confidence value for the integrated_fitCons_score:	0 - High confidence values (p<~.003), 1 - Likely Significant (p<.05), 2 - Likely Informative (p<.25), 3 - Best estimate (p>=.25)
GERP++_RS: GREP++ "rejected substitutions" (RS) score, higher scores are more deleterious
GERP++_RS_rankscore: the rank of the GERP_RS score among all GERP_RS scores in genome
phyloP100way_vertebrate: Phylogenetic p-values for 100 vertebrate species, higher scores are more deleterious
phyloP100way_vertebrate_rankscore: the rank of the phyloP100way_vertebrate score among all phyloP100way_vertebrate scores in genome
phyloP20way_mammalian:  Phylogenetic p-values for 20 mammalian species, higher scores are more deleterious
phyloP20way_mammalian_rankscore: the rank of the phyloP20way_mammalian score among all phyloP20way_mammalian scores in genome
phastCons100way_vertebrate: PhastCons score for 100 vertebrate species, higher scores are more deleterious
phastCons100way_vertebrate_rankscore: the rank of the phastCons100way_vertebrate score among all phastCons100way_vertebrate scores in genome
phastCons20way_mammalian: PhastCons score for 20 mammalian species, higher scores are more deleterious
phastCons20way_mammalian_rankscore: the rank of the phastCons20way_mammalian score among all phastCons20way_mammalian scores in genome
SiPhy_29way_logOdds: SiPhy log odds score for 29 species, higher scores are more deleterious
SiPhy_29way_logOdds_rankscore: the rank of the SiPhy_29way_logOdds score among all SiPhy_29way_logOdds scores in genome
Interpro_domain
GTEx_V6p_gene: Ensembl gene ID of the eQTL associated with
GTEx_V6p_tissue: the tissue expression data in which the eQTL was detected
REVEL: 
MCAP: 
regsnp_fpr: 
regsnp_disease: 
regsnp_splicing_site: 
dbscSNV_ADA_SCORE: 
dbscSNV_RF_SCORE: 
InterVar_automated: 
PVS1
PS1
PS2
PS3
PS4
PM1
PM2
PM3
PM4
PM5
PM6
PP1
PP2
PP3
PP4
PP5
BA1
BS1
BS2
BS3
BS4
BP1
BP2
BP3
BP4
BP5
BP6
BP7



### [Kaviar](http://db.systemsbiology.net/kaviar/)

Kaviar contains 162 million SNV sites (including 25M not in dbSNP) and incorporates data from 35 projects encompassing 77,781 individuals (13.2K whole genome, 64.6K exome)


### [PROVEAN](http://provean.jcvi.org/index.php)

PROVEAN (Protein Variation Effect Analyzer) is a software tool which predicts whether an amino acid substitution or indel has an impact on the biological function of a protein.
